File: In C++/SelectionSort.cpp
#include <iostream>

void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; ++i) {
        int minIndex = i;

        // Find the index of the minimum element in the unsorted part of the array
        for (int j = i + 1; j < n; ++j) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }

        // Swap the minimum element with the first element of the unsorted part
        if (minIndex != i) {
            std::swap(arr[i], arr[minIndex]);
        }
    }
}

int main() {
    int arr[] = {64, 25, 12, 22, 11};
    int n = sizeof(arr) / sizeof(arr[0]);

    std::cout << "Original Array: ";
    for (int i = 0; i < n; ++i) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    selectionSort(arr, n);

    std::cout << "Sorted Array: ";
    for (int i = 0; i < n; ++i) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}


//Selection Sort in C++

File: In Java/Selection Sort/Selection_sort.java


public class Selection_sort {//👈Your File name and class Name should be same
    public static void main(String[] args) {
        int arr[] = {3,2,1,0};
        int n = arr.length;

        solution(arr, n);

        for(int i=0;i<n;i++){
            System.out.print(arr[i]+" ");
        }
        
    }
   static void solution(int arr [],int n){
      
    for(int i=0;i<n;i++){
        int min_index = i;
        for(int j=i+1;j<n;j++){
            if(arr[j]<arr[min_index]){
                min_index = j;
            }
        }

        //Swap the min index because our min index is smallest.
        int temp = arr[i];
        arr[i]= arr[min_index];
        arr[min_index ]= temp;

    }

   }
   //Time Complexity is O(n^2)
   //Selection sort is a relatively simple algorithm to understand and implement. However, it is not very efficient for sorting large lists. There are other sorting algorithms, such as quicksort and merge sort, that have a better time complexity and are therefore more efficient for sorting large lists.
       


    
    
}
